### DISCUSSION TITLE
Gallery

### DISCUSSION THREAD

**Discussion by torvenkat** [binokochumolvarghese/lightbi-hugo#51] [Ideas]
Hi, Really good and clean theme and thanks for your efforts and sharing it. 
Will it be possible to implement a gallery option?  Something along the lines of https://www.liwen.id.au/heg/
Thanks.

**Comment by binokochumolvarghese** [comment]
Hi, Thanks to the kind words. I will try to implement that and hopefully I can release it soon.

**Reply by tkisch** [reply]
Hi Bino

Also from my side many thanks for the wonderful theme. I am also trying to implement a gallery with Lighbox functionality. Unfortunately - as an absolute beginner - without any success.

I am currently trying to integrate the following:

https://hugocodex.org/add-ons/image-gallery/
https://hugocodex.org/add-ons/lightbox/

I can get the gallery to work.
The lightbox, however, does not. 
It opens, but at the bottom of the page and without any navigation button to change the photos.

Do you have any advice on how to solve this?

Best regards
Tobias

**Comment by torvenkat** [comment]
This one looks stunning!  Not sure how difficult it is to implement. https://nanogallery2.nanostudio.org/

Here are some additional notes on the above for Hugo. https://mertbakir.gitlab.io/hugo/nanogallery2-with-hugo/

**Comment by binokochumolvarghese** [comment]
@torvenkat @tkisch 

I am currently working on implementing a gallery feature and would like your input on the best structure for handling multiple images. Should we:

- Provide an option to display all images alongside a Lightbox view for better navigation?
- Or display the first four images with a "View More" button to reveal the rest?

Which approach do you think would offer a better user experience?

**Reply by tkisch** [reply]
I like the this approach:
https://github.com/mfg92/hugo-shortcode-gallery
(Unfortunately i wasn't able to get it running)

**Comment by binokochumolvarghese** [comment]
You can view the beta version of image gallery [here](https://lightbi-beta-demo.netlify.app/en/post/2020/image-gallery/).
